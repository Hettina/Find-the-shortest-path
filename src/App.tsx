import React, { Component } from 'react';
import Dijkstra from './Dijkstra/dijkstra'
import './App.css';
const diagram = require('./assets/diagram.jpg');

class App extends Component<ComponentProps, ComponentState> {
    constructor(props: any) {
        super(props);
        this.state = {
            startNode: '',
            endNode: ''
        };
    }

    handleOnChange= (event:React.FormEvent<HTMLInputElement>) : void =>{
        event.preventDefault();
        let value = event.currentTarget.value.toUpperCase();
        // @ts-ignore
        this.setState({[event.target.name]:  value});
    };


    render() {
        const { startNode, endNode } = this.state;
        return (
            <div className="App">
                {<img src={diagram} className="diagram" alt="graph diagram" />}
                <h1 className="App-header">
                    Find the Shortest Path
                </h1>

                <div>
                    <label htmlFor="startNode">Please enter a start point</label>
                    <input
                        type="text"
                        name="startNode"
                        className="startNode"
                        value={startNode}
                        onChange={(event) => this.handleOnChange(event)}
                    />
                    <label htmlFor="endNode">Please enter an end point</label>
                    <input
                        type="text"
                        name="endNode"
                        value={endNode}
                        onChange={(event) => this.handleOnChange(event)}
                    />
                    {(startNode!=='' && endNode!=='') && (
                        <Dijkstra start={startNode} end={endNode}/>
                    )}
                </div>
            </div>
        );
    }
}

interface ComponentProps { }
interface ComponentState {
    startNode: string
    endNode: string
}

export default App;
