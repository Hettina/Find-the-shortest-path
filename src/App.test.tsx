import React from 'react';
import { render, cleanup, fireEvent} from 'react-testing-library';
import App from './App';

afterEach(cleanup);

describe('Test Dijkstra component', () => {
  it('renders app', () => {
    const container = render(<App />);
    expect(container).toMatchSnapshot()
  });

  it('should not display result paragraph in snapshot when only one input is added', async () => {
    const {container} = render(<App/>);
    // @ts-ignore
    const input: HTMLInputElement  = container.querySelector('input.startNode');

    fireEvent.change(input, {target: {value: 'h'}});
    expect(container).toMatchSnapshot()
  });

  it('should fire handleOnChange when something is typed in the input', async () => {
      const {container} = render(<App/>);
      // @ts-ignore
      const input: HTMLInputElement  = container.querySelector('input.startNode');

      fireEvent.change(input, {target: {value: 'h'}});
      expect(input.value).toBe('H')
  });

});
