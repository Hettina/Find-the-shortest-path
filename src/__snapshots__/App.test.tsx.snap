// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Dijkstra component renders app 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="App"
      >
        <img
          alt="graph diagram"
          class="diagram"
          src="diagram.jpg"
        />
        <h1
          class="App-header"
        >
          Find the Shortest Path
        </h1>
        <div>
          <label
            for="startNode"
          >
            Please enter a start point
          </label>
          <input
            class="startNode"
            name="startNode"
            type="text"
            value=""
          />
          <label
            for="endNode"
          >
            Please enter an end point
          </label>
          <input
            name="endNode"
            type="text"
            value=""
          />
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="App"
    >
      <img
        alt="graph diagram"
        class="diagram"
        src="diagram.jpg"
      />
      <h1
        class="App-header"
      >
        Find the Shortest Path
      </h1>
      <div>
        <label
          for="startNode"
        >
          Please enter a start point
        </label>
        <input
          class="startNode"
          name="startNode"
          type="text"
          value=""
        />
        <label
          for="endNode"
        >
          Please enter an end point
        </label>
        <input
          name="endNode"
          type="text"
          value=""
        />
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Test Dijkstra component should not display result paragraph in snapshot when only one input is added 1`] = `
<div>
  <div
    class="App"
  >
    <img
      alt="graph diagram"
      class="diagram"
      src="diagram.jpg"
    />
    <h1
      class="App-header"
    >
      Find the Shortest Path
    </h1>
    <div>
      <label
        for="startNode"
      >
        Please enter a start point
      </label>
      <input
        class="startNode"
        name="startNode"
        type="text"
        value="H"
      />
      <label
        for="endNode"
      >
        Please enter an end point
      </label>
      <input
        name="endNode"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
